zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -608,7 +610,11 @@ public synchronized void startup() {
         registerMetrics();
 
         setState(State.RUNNING);
+
         requestPathMetricsCollector.start();
+
+        localSessionEnabled = sessionTracker instanceof UpgradeableSessionTracker
```

Comment:
this 'instanceof' looks like a 'code smell'
what about adding a isLocalSessionsEnabled method to the base class (SessionTracker) with a default return value of 'false' ?

Reply:
Yeap, makes sense. changes made.